app/Libraries/Blog.php

```diff
@@ -0,0 +1,280 @@
+<?php namespace App\Libraries;
+
+use App\Entities\Post;
+use App\Exceptions\BlogException;
+use CodeIgniter\Exceptions\PageNotFoundException;
+use Config\Blog as BlogConfig;
+use League\CommonMark\CommonMarkConverter;
+
+/**
+ * Class Blog
+ *
+ * Handles reading and parsing content posts.
+ * Posts are markdown files versioned with the
+ * site repo, with a simple yaml header section
+ * to define meta data.
+ */
+class Blog
+{
+    /**
+     * @var BlogConfig
+     */
+    protected $config;
+
+    public function __construct()
+    {
+        $this->config = config(BlogConfig::class);
+    }
```

Comment:
```suggestion
    public function __construct(BlogConfig $config = null)
    {
        $this->config = $config ?? config(BlogConfig::class);
    }
```
Let's inject here to make things like testing easier down the road.